src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -818,12 +819,16 @@ private void maybeWarn(ResultSetBuilder result, QueryOptions options)
     {
         if (!options.isClientTrackWarningsEnabled())
             return;
+        ColumnFamilyStore store = cfs();
+        if (store != null)
+            store.metric.clientReadSize.update(result.getSize());
         if (result.shouldWarn(options.getClientLargeReadWarnThresholdKb()))
         {
             String msg = String.format("Read on table %s has exceeded the size warning threshold of %,d kb", table, options.getClientLargeReadWarnThresholdKb());
             ClientWarn.instance.warn(msg + " with " + loggableTokens(options));
             logger.warn("{} with query {}", msg, asCQL(options));
-            cfs().metric.clientReadSizeWarnings.mark();
```

Comment:
Was it possible for this to be `null` before and we weren't checking, or is it only now possible and hence the check?

Reply:
there is one code path (not this) which hit a null (called from unit test), so I retroactively updated all paths to handle null

Reply:
Gotcha...I'm always a little worried when we relax a constraint like this, but if the tests really can't be set up to avoid it, that's ok.